src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -432,14 +432,14 @@ async def build_pex(
             description_of_origin=request.requirements.file_path_description_of_origin,
         )
 
-        requirements_file_digest_contents = await Get(DigestContents, PathGlobs, globs)
-
-        metadata = LockfileMetadata.from_lockfile(
-            requirements_file_digest_contents[0].content,
-            request.requirements.file_path,
-            resolve_name,
-        )
-        _validate_metadata(metadata, request, request.requirements, python_setup)
+        if python_setup.invalid_lockfile_behavior in {InvalidLockfileBehavior.warn, InvalidLockfileBehavior.error}:
```

Comment:
You need to do the same thing down at line `451` (the logic for `LockfileContent` requirements too).

It may be worth refactoring this if you can figure out how to make it more DRY